cmd/manager/main.go

```diff
@@ -4,140 +4,152 @@ import (
-	var clusterOperatorName string
+	var (
+		clusterOperatorName string
+		tlsKeyPath          string
+		tlsCertPath         string
+		version             bool
+	)
 	flag.StringVar(&clusterOperatorName, "clusterOperatorName", "", "the name of the OpenShift ClusterOperator that should reflect this operator's status, or the empty string to disable ClusterOperator updates")
+	flag.StringVar(&defaults.Dir, "defaultsDir", "", "the directory where the default CatalogSources are stored")
+	flag.BoolVar(&version, "version", false, "displays marketplace source commit info.")
+	flag.StringVar(&tlsKeyPath, "tls-key", "", "Path to use for private key (requires tls-cert)")
+	flag.StringVar(&tlsCertPath, "tls-cert", "", "Path to use for certificate (requires tls-key)")
 	flag.Parse()
 
 	// Check if version flag was set
-	if *version {
-		fmt.Print(sourceCommit.String())
-		// Exit immediately
+	if version {
+		logrus.Infof(sourceCommit.String())
 		os.Exit(0)
 	}
 
 	// set TLS to serve metrics over a secure channel if cert is provided
 	// cert is provided by default by the marketplace-trusted-ca volume mounted as part of the marketplace-operator deployment
-	var useTLS bool
-	if *tlsCertPath != "" && *tlsKeyPath == "" || *tlsCertPath == "" && *tlsKeyPath != "" {
-		log.Warn("both --tls-key and --tls-crt must be provided for TLS to be enabled, falling back to non-https")
-	} else if *tlsCertPath == "" && *tlsKeyPath == "" {
-		log.Info("TLS keys not set, using non-https for metrics")
-	} else {
-		log.Info("TLS keys set, using https for metrics")
-		useTLS = true
-	}
-	err := metrics.ServePrometheus(useTLS, *tlsCertPath, *tlsKeyPath)
+	err := metrics.ServePrometheus(tlsCertPath, tlsKeyPath)
 	if err != nil {
-		log.Fatalf("failed to serve prometheus metrics: TLS enabled %d: %s", useTLS, err)
+		logrus.Errorf("failed to serve prometheus metrics: %s", err)
+		os.Exit(1)
 	}
 
 	namespace, err := k8sutil.GetWatchNamespace()
 	if err != nil {
-		log.Fatalf("failed to get watch namespace: %v", err)
+		logrus.Errorf("failed to get watch namespace: %v", err)
+		os.Exit(1)
 	}
 
 	// Get a config to talk to the apiserver
 	cfg, err := config.GetConfig()
 	if err != nil {
-		log.Fatal(err)
+		logrus.Error(err)
+		os.Exit(1)
 	}
 
 	// Set OpenShift config API availability
 	err = configv1.SetConfigAPIAvailability(cfg)
 	if err != nil {
-		log.Fatal(err)
+		logrus.Error(err)
+		os.Exit(1)
 	}
 
 	// Create a new Cmd to provide shared dependencies and start components
 	// Even though we are asking to watch all namespaces, we only handle events
 	// from the operator's namespace. The reason for watching all namespaces is
 	// watch for CatalogSources in targetNamespaces being deleted and recreate
 	// them.
-	mgr, err := manager.New(cfg, manager.Options{Namespace: ""})
+	mgr, err := manager.New(cfg, manager.Options{
+		Namespace:              "",
+		HealthProbeBindAddress: ":8080",
+		MetricsBindAddress:     "0",
```

Comment:
This is an artifact of controller-runtime making a breaking change from defaulting to port 8383 to 8080, which conflicts with the port that we expose for readiness and liveness probes. Setting this value to `0` disables metrics, so we'll need to double-check if that's the behavior we want, vs. updating the port where we serve our health checks.

Reply:
This is for the metrics that controller runtime exposes, which to my understanding, weren't being exposed anyways when I poke around a <4.8 marketplace instance. I can play around with getting these metrics enabled and exposed.